src/main/java/se/iths/jakartaeelab1/exceptions/ConstraintViolationExceptionMapper.java

```diff
@@ -0,0 +1,56 @@
+package se.iths.jakartaeelab1.exceptions;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.ConstraintViolationException;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+
+import java.util.Set;
+
+@Provider
+public class ConstraintViolationExceptionMapper implements ExceptionMapper<ConstraintViolationException> {
+    @Context
+    private UriInfo uriInfo;
+
+    @Override
+    public Response toResponse(ConstraintViolationException exception) {
+        Set<ConstraintViolation<?>> constraintViolations = exception.getConstraintViolations();
+        JsonObjectBuilder jsonObjectBuilder = Json.createObjectBuilder()
+                .add("host", uriInfo.getAbsolutePath().getHost())
+                .add("resource", uriInfo.getAbsolutePath().getPath())
+                .add("title", "Validation Errors");
+
+        JsonArrayBuilder jsonArrayBuilder = Json.createArrayBuilder();
+        for (ConstraintViolation<?> constraint : constraintViolations) {
+            String message = constraint.getMessage();
+            String propertyPath = constraint.getPropertyPath().toString();
+            String[] pathParts = propertyPath.split("\\.");
+
+            //seeing if path has at least 3 parts to avoid IndexOutOfBoundsException
+            String violatedField = pathParts.length > 2 ? pathParts[2] : "unknown";
```

Comment:
Why do we need to do this check? The code works and were getting the formatting we want on the validation-errors. But the code I suggested did this aswell. Can you explain a bit why you changed it? :)

Reply:
I just thought I'd add something to ensure robustness, just a small safety check for the property path structure to avoid IndexOutOfBoundsException and for it to set a default field name if the structure isn't as expected. Not needed for the program to work properly though like you said 👍